### PR TITLE
SUP-42 handle skadn imp->ext

### PR DESCRIPTION
**Pull request by eric-kansas**
update prebid openrtb_ext to accept SKADN request object; 
update tapjoy utilized adapters to filter SKADN for each dsp and add to imp ext

### CODE REVIEW

File: adapters/openrtb_util.go [hunk 1]
```diff
@@ -173,3 +174,28 @@ func copyFormats(sizes []openrtb.Format) []openrtb.Format {
...
+
+	return openrtb_ext.SKADN{
+		Version:    prebidExt.SKADN.Version,
+		SourceApp:  prebidExt.SKADN.SourceApp,
+		SKADNetIDs: filterArrayWithMap(prebidExt.SKADN.SKADNetIDs, filterMap),
+	}
+}
+
+// filterArrayWithMap -- Added by Tapjoy to handle SKADN DSP filtering
+// returns a subset elemtns of arr whose keys were in filterMap
```
Comment (umutdemirel) [inline adapters/openrtb_util.go:193]: fix typo

File: adapters/rubicon/rubicon.go [hunk 2]
```diff
@@ -33,6 +33,10 @@ const (
 	badvLimitSize = 50
 )
 
+var rubiconSKADNetIDs = map[string]bool{
+	"abc": true,
```
Comment (umutdemirel) [inline adapters/rubicon/rubicon.go:37]: Can you confirm if Rubicon doesn't have an skadnetid for now, so this is a placeholder?

**Review by umutdemirel — APPROVED** [review]
Code looks good to me w/some comments.
Also, it'd be nice to add tests for `FilterPrebidSKADNExt` and `filterArrayWithMap` functions.

File: adapters/rubicon/rubicon.go (hunk 2 above)
Comment (eric-kansas) [reply adapters/rubicon/rubicon.go:37]: @punktea00

**Pull request merged**